fuzz.go

```diff
@@ -205,6 +208,9 @@ func (f *Fuzzer) SkipFieldsWithPattern(pattern *regexp.Regexp) *Fuzzer {
 // golang :/ ) Intended for tests, so will panic on bad input or unimplemented
 // fields.
 func (f *Fuzzer) Fuzz(obj interface{}) {
+	f.fuzzLock.Lock()
```

Comment:
I think the intended solution to this concern was for everything dependent to go into the fuzzContext. I'm not completely sure how to fix this without changing the sequence of random numbers it uses.

Reply:
But there we do recurse so we can't lock. The more involved approach would be to wrap with something that locks